### PR TITLE
[DC-813] Update Temporal Data Point CR to New Base Class & Correct Pi…

### PR DESCRIPTION
**Pull request by amcgrenera-vumc**
…peline Step

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py [hunk 1]
```diff
@@ -32,112 +37,214 @@
...
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Returns queries to updated end dates, end dates should not be prior to any start date'
```
Comment (dcampbell-vumc) [inline data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py:157]: nit:
```suggestion
            'Returns queries to update end dates, end dates should not be prior to any start date'
```

**Review by dcampbell-vumc — CHANGES_REQUESTED** [review]
Do you think we should add a sandboxing query? I commented a sandbox query suggestion but it hasn't been tested. I am not sure if it would be too difficult to add something like that or not.

File: data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py [hunk 2]
```diff
@@ -32,112 +37,214 @@
...
+  SELECT
+    *
+  FROM
+    `{project_id}.{dataset_id}.{table}`
+  WHERE
+    NOT {table_end_date} < {table_start_date}) r
+ON
+  l.{TABLE}_id = r.{TABLE}_id
+""")
+
```
[the remark this answers is not in the document]
Comment (amcgrenera-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py:55]: Good question! Let me ask in the eng channel, I would think we should and this seems like an older cleaning rule before that was put in place, but I'll double check!

[the remark this answers is not in the document]
Comment (amcgrenera-vumc) [reply data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py:55]: Krishna said yes, so I will be adding!

**Comment by amcgrenera-vumc** [comment]
@dcampbell-vumc now updated with sandboxing!

File: data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py [hunk 3]
```diff
@@ -29,115 +34,255 @@
...
-    for table in table_dates:
-        fields = resources.fields_for(table)
-        # Generate column expressions for select
-        col_exprs = [
-            'r.' +
-            field['name'] if field['name'] == table_dates[table][1] else 'l.' +
-            field['name'] for field in fields
-        ]
-        cols = ', '.join(col_exprs)
+SANDBOX_BAD_END_DATES = JINJA_ENV.from_string("""
```
Comment (dcampbell-vumc) [inline data_steward/cdr_cleaner/cleaning_rules/temporal_consistency.py:43]: looks great!

**Review by dcampbell-vumc — APPROVED** [review]
This looks good to me! Thanks for adding the sandboxing class!

**Pull request merged**